src/analytics/helpers.js

```diff
@@ -0,0 +1,94 @@
+import moment from 'moment'
+import { links } from 'config'
+import formatFacetNames from 'helpers/formatFacetNames'
+import formatFilterName from 'helpers/formatFilterName'
+import getReadable from 'helpers/getReadable'
+
+const datasetOptionsKeys = ['aggregate_by', 'scale_by', 'quantile_normalize']
+
+// adds a custom event to GA4
+// https://developers.google.com/analytics/devguides/collection/ga4/events?client_type=gtag
+export const event = (eventName, value = {}, nonInteraction = false) => {
+  window.gtag('event', eventName, {
+    ...value,
+    non_interaction: nonInteraction
+  })
+}
+
+// formats a given string to snake_case for event parameters
+export const getSnakeCase = (str) =>
+  str.replace(/([a-z])([A-Z])/g, '$1_$2').toLowerCase()
+
+export const getDatasetOptionsChanges = (dataset, regeneratedDataset) => {
+  const changes = datasetOptionsKeys.map((key) => {
+    return `${getReadable(key, dataset[key])} -> ${getReadable(
+      key,
+      regeneratedDataset[key]
+    )}`
+  })
+  return changes.join('|')
+}
+
+export const getDatasetState = (dataset) =>
+  getReadable('expires_on', moment(dataset.expires_on).isBefore(Date.now()))
```

Comment:
I've added the `expires_on` key to the config 's`readableBooleans` for `getReadable` (replacing the previous `isExpired` parameter). This key returns either `'Expired'` or `'Valid'` as fixed values. Let me know what you think!

```
export const readableBooleans = {
  expires_on: ['Expired', 'Valid'],
  quantile_normalize: ['Not skipped', 'Skipped'],
  quant_sf_only: ['rna-seq', 'normalized']
}
```